airflow/models/pool.py

```diff
@@ -107,3 +109,33 @@ def open_slots(self, session):
         Returns the number of slots open at the moment
         """
         return self.slots - self.occupied_slots(session)
+
+    @provide_session
+    def slots_stats(self, session) -> Dict:
+        from airflow.models.taskinstance import TaskInstance  # Avoid circular import
+
+        states_filter = STATES_TO_COUNT_AS_RUNNING | frozenset([State.RUNNING, State.QUEUED])
```

Comment:
Aren't running and queued already in that?

Reply:
It is, this is to safe guard future changes that will add or remove states from `STATE_TO_COUNT_AS_RUNNING` set. The naming of that variable doesn't seem to guarantee it will only contain queued and running going forward. I can certainly remove this safe guard code if that's not the case.